internal/wasm/interpreter/interpreter.go

```diff
@@ -619,20 +641,18 @@ func (ce *callEngine) callNativeFunc(ctx *wasm.ModuleContext, f *compiledFunctio
 				if offset >= uint64(len(table.Table)) {
 					panic(wasm.ErrRuntimeInvalidTableAccess)
 				}
-				tableElement := table.Table[offset]
-				// Type check.
-				if uint64(tableElement.FunctionTypeID) != op.us[1] {
-					if tableElement.FunctionTypeID == wasm.UninitializedTableElementTypeID {
-						panic(wasm.ErrRuntimeInvalidTableAccess)
-					}
+				targetCompiledFunction := (*compiledFunction)(unsafe.Pointer(table.Table[offset]))
```

Comment:
Does it work well after GC moved the address of compiledFunctions?

Reply:
go's GC doesn't do compaction or relocation, so as long as the *compilerFunction is alive, this is safe

Reply:
oh, I see. is that behavior just true for the current implementation, or guaranteed by spec?  But anyway no problem if you're sure. I asked just because I see the warning for this line on VSCode and found this document that warns it https://pkg.go.dev/unsafe#:~:text=A%20uintptr%20is%20an,object%20from%20being%20reclaimed.

Reply:
GC's behavior is not something written in language's spec, but you can see https://github.com/golang/go/blob/master/src/runtime/mgc.go#L9

Reply:
but any way on second though, I think I will use `interface` instead of uintptr for safety in the subsequent PR